### PR TITLE
chore: add helper tooltip to dashboard highlight cards

### LINKED ISSUE #1806: Feature: Helper tooltips for dashboards
### Type of feature

🍕 Feature

### Current behavior

I've gotten feedback from users (even maintainers) that open up our dashboards but aren't sure what they are seeing and why it is helpful.

### Suggested solution

I think a small step forward towards educating users could be adding some context through tooltips that display when the user hovers over a KPI title. 

The copy on this mock is not great, but it's a way of showing what can be done:

image

We can continue to tackle this issue by creating our own documentation for OSS metrics and how to measure a project / use our platform, and linking that content in the helper tooltips. That way if a user asks themselves "why is this useful?" they can just hover over the title and get more info on it.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR adds helper tooltip to dashboard highlights cards

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #1806

## Mobile & Desktop Screenshots/Recordings

image
image

image

image

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
This is just a PR to add helper tooltips to the dashboard cards, i'll apply tooltips related to the table data in https://github.com/open-sauced/app/pull/2200. cc: @BekahHW

**Comment by OgDev-01** [comment]
Made a tiny update to the y-axis padding of the tooltip
image
I think this looks a lot better than the first

**Comment by BekahHW** [comment]
@OgDev-01 is there a way to make the tooltip appear in the same place in every card? If you look at your screenshots, the one on spam is outside of the card.

File: components/molecules/HighlightCard/highlight-card.tsx [hunk 1]
```diff
@@ -154,7 +160,17 @@ const HighlightCard: React.FC<HighlightCardProps> = ({
               {getIcon(icon)}
             </div>
             {/* Label: Text */}
-            <div className="text-sm text-slate-600   leading-none">{label ? label : "Label"}</div>
+            <div className={clsx("text-sm text-slate-600 leading-none", tooltip && "cursor-pointer")}>
```
Comment (jpmcb) [inline components/molecules/HighlightCard/highlight-card.tsx:163]: Not sure `cursor-pointer` is the type of hover cursor we want: it makes me think it's a link but clicking it doesn't do anything.

Comment (nickytonline) [reply components/molecules/HighlightCard/highlight-card.tsx:163]: Agree with @jpmcb here. No need for this cursor. When they hover, it will just appear.

**Comment by OgDev-01** [comment]
>@OgDev-01 is there a way to make the tooltip appear in the same place in every card? If you look at your screenshots, the one on spam is outside of the card.

Thank you for bringing this up. The tooltip placement you've noticed is intentional and aligns with the default behavior of tooltips in the Radix. The tooltips are designed to appear relative to their parent elements, which in this case is the card title.

The `Spam` card looks a bit outside the card as it is position at the center in relative on the parent text( the `Spam` word). not sure if there is a way to skip this behaviour of the tooltip.

File: components/molecules/HighlightCard/highlight-card.tsx [hunk 2]
```diff
@@ -154,7 +160,17 @@ const HighlightCard: React.FC<HighlightCardProps> = ({
...
+            <div className={clsx("text-sm text-slate-600 leading-none")}>
+              {tooltip ? (
+                <Tooltip className="w-44 py-3 text-center" content={tooltip}>
+                  {label}
+                </Tooltip>
+              ) : label ? (
+                label
+              ) : (
+                "Label"
+              )}
```
Comment (nickytonline) [inline components/molecules/HighlightCard/highlight-card.tsx:172]: refactor (blocking): The `tooltip` and `label` props are both optional. It's unclear why if there is no `tooltip` or `label` prop that the word `Label` is returned.

```suggestion
              {tooltip && label ? (
                <Tooltip className="w-44 py-3 text-center" content={tooltip}>
                  {label}
                </Tooltip>
              ) : label ? (
                label
              ) : (
                "Label"
              )}
```

File: components/molecules/HighlightCard/highlight-card.tsx [hunk 3]
```diff
@@ -154,7 +160,17 @@ const HighlightCard: React.FC<HighlightCardProps> = ({
               {getIcon(icon)}
             </div>
             {/* Label: Text */}
-            <div className="text-sm text-slate-600   leading-none">{label ? label : "Label"}</div>
+            <div className={clsx("text-sm text-slate-600 leading-none")}>
+              {tooltip && lable ? (
```
Comment (brandonroberts) [inline components/molecules/HighlightCard/highlight-card.tsx:164]: typo is causing the build failure

```suggestion
              {tooltip && label ? (
```

**Review by brandonroberts — APPROVED** [review]

**Review by nickytonline — APPROVED** [review]
🚀

**Pull request merged**